lua/ltex-ls/cache.lua

```diff
@@ -17,19 +17,30 @@ end
 ---@param client vim.lsp.Client
 ---@return string path The path to the cache
 local function get_cache_path(client)
-  return vim.fs.joinpath(client.root_dir, M.CACHE_FNAME)
+  if client.config.root_dir then
+    return M.joinpath(client.config.root_dir, M.CACHE_FNAME)
+  else 
+    return nil
+  end
+end
+function M.joinpath(...)
```

Comment:
Hey, I think it is better to stick with the "stdlib" version of the function here (that'll avoid some unnecessary maintaining on our side).

Reply:
it wasn't available for some reason in my version of nvim 0.9.x iirc